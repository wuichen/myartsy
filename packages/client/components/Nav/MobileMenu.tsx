import { garamond } from "Assets/Fonts"
import React from "react"
import styled, { StyledFunction } from "styled-components"
import StyledTextLink from "../TextLink"

interface Props {
  navHeight?: string
}

export const MobileMenu: React.SFC<Props> = props => {
  return (
    <MobileMenuContainer {...props}>
      <StyledTextLink color="white" href="/">
        Home
      </StyledTextLink>
      <StyledTextLink color="white" href="/institutions">
        Museums
      </StyledTextLink>
      <StyledTextLink color="white" href="/artists">
        Artists
      </StyledTextLink>
      <StyledTextLink color="white" href="/articles">
        Articles
      </StyledTextLink>
      <StyledTextLink color="white" href="/shows">
        Show Guide
      </StyledTextLink>
      <StyledTextLink color="white" href="/art-fairs">
        Art Fairs
      </StyledTextLink>
      <StyledTextLink color="white" href="/galleries">
        Galleries
      </StyledTextLink>
      <StyledTextLink color="white" href="/auctions">
        Auctions
      </StyledTextLink>
    </MobileMenuContainer>
  )
}

MobileMenu.defaultProps = {
  navHeight: "0px",
}

const Div: StyledFunction<Props & React.HTMLProps<HTMLDivElement>> = styled.div
const MobileMenuContainer = Div`
  ${garamond("s19")};
  box-sizing: border-box;
  position: absolute;
  z-index: 1;
  width: 100%;
  background-color: black;
  left: 0;
  padding: 20px;
  margin-top: ${props => props.navHeight};
  ${StyledTextLink} {
    display: inline-block;
    width: 50%;
    line-height: 2;
  }
`
