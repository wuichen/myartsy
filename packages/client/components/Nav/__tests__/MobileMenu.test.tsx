import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { MobileMenu } from "../MobileMenu"

it("renders a mobile menu properly", () => {
  const menu = renderer.create(<MobileMenu />).toJSON()
  expect(menu).toMatchSnapshot()
})
