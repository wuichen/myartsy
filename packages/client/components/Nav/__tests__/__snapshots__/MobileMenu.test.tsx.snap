// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a mobile menu properly 1`] = `
.c2 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 15px;
  line-height: 1.25em;
  -webkit-font-smoothing: antialiased;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 19px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  box-sizing: border-box;
  position: absolute;
  z-index: 1;
  width: 100%;
  background-color: black;
  left: 0;
  padding: 20px;
  margin-top: 0px;
}

.c0 .c1 {
  display: inline-block;
  width: 50%;
  line-height: 2;
}

<div
  className="c0"
>
  <a
    className="c1 c2"
    href="/"
    target="_self"
  >
    Home
  </a>
  <a
    className="c1 c2"
    href="/institutions"
    target="_self"
  >
    Museums
  </a>
  <a
    className="c1 c2"
    href="/artists"
    target="_self"
  >
    Artists
  </a>
  <a
    className="c1 c2"
    href="/articles"
    target="_self"
  >
    Articles
  </a>
  <a
    className="c1 c2"
    href="/shows"
    target="_self"
  >
    Show Guide
  </a>
  <a
    className="c1 c2"
    href="/art-fairs"
    target="_self"
  >
    Art Fairs
  </a>
  <a
    className="c1 c2"
    href="/galleries"
    target="_self"
  >
    Galleries
  </a>
  <a
    className="c1 c2"
    href="/auctions"
    target="_self"
  >
    Auctions
  </a>
</div>
`;
